In [1]:
import warnings
warnings.filterwarnings('ignore')
warnings.filterwarnings('always')

In [2]:
# Importación librerías
import pandas as pd
import numpy as np

<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen 

In [3]:
# Carga de datos de archivo .csv
dataTraining = pd.read_csv('https://raw.githubusercontent.com/davidzarruk/MIAD_ML_NLP_2023/main/datasets/dataTrain_carListings.zip')
dataTesting = pd.read_csv('https://raw.githubusercontent.com/davidzarruk/MIAD_ML_NLP_2023/main/datasets/dataTest_carListings.zip', index_col=0)

<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()


In [4]:
# Visualización datos de entrenamiento
dataTraining.head()

Price  Year  Mileage State       Make      Model
0  34995  2017     9913    FL       Jeep   Wrangler
1  37895  2015    20578    OH  Chevrolet   Tahoe4WD
2  18430  2012    83716    TX        BMW      X5AWD
3  24681  2014    28729    OH   Cadillac  SRXLuxury
4  26998  2013    64032    CO       Jeep   Wrangler

De que trata el dataset:

Price: precio del vehiculo usado

Year: año de fabricacion, salida al mercado

Mileage: millas recorridas (kilometraje)

Make: marca del fabricante

Model: modelo de vehiculo

In [5]:
#verificacion de datos faltantes
valores_nulos_por_columna = dataTraining.isnull().sum()
valores_nulos_por_columna

Price      0
Year       0
Mileage    0
State      0
Make       0
Model      0
dtype: int64

No se registran datos faltantes

In [6]:
# tipos de datos


tipos=dataTraining.dtypes
print(tipos)

Price       int64
Year        int64
Mileage     int64
State      object
Make       object
Model      object
dtype: object


Se tienen 3 caracteristicas numericas y 3 categoricas, verifiquemos la cantidad de clases de cada categoria:

In [7]:
#conteo de datos categoricos:
#conteo de categorias en la variable Class
conteostate = dataTraining['State'].value_counts()

display(conteostate)
len(conteostate)

State
 TX    45918
 CA    36534
 FL    33759
 GA    18182
 NC    17930
 IL    16793
 VA    15894
 PA    13039
 NY    12447
 NJ    12132
 OH    11364
 WA    10900
 CO    10875
 AZ    10749
 TN     9274
 MA     8586
 MD     8245
 MO     8029
 IN     7850
 KY     6688
 AL     6283
 WI     5614
 MN     5415
 MI     5327
 OK     5096
 CT     4942
 OR     4907
 UT     4854
 SC     4815
 LA     3847
 KS     3704
 NV     3332
 AR     3128
 MS     3043
 NH     2744
 NE     2522
 IA     2325
 NM     2280
 ID     1736
 DE     1221
 HI     1181
 MT      997
 ME      965
 RI      833
 AK      813
 ND      681
 WV      666
 SD      621
 VT      579
 WY      335
 DC        6
Name: count, dtype: int64

51

In [8]:
#conteo de datos categoricos:

conteoMake = dataTraining['Make'].value_counts()

display(conteoMake)
len(conteoMake)

Make
Ford             62899
Chevrolet        58383
Toyota           45941
Honda            33191
Jeep             24369
GMC              20834
Kia              16352
Dodge            16159
Hyundai          15057
Lexus            13664
BMW              12326
Volkswagen       11110
Nissan           10569
Chrysler          9046
Mercedes-Benz     7575
Subaru            6791
Cadillac          5414
Buick             4842
Ram               3611
MINI              2766
Land              2713
INFINITI          2406
Acura             2403
Mazda             2332
Lincoln           1786
Volvo             1710
Audi              1423
Mitsubishi        1272
Porsche            980
Scion              651
Jaguar             324
Pontiac            308
FIAT               272
Mercury            271
Tesla              148
Bentley             57
Suzuki              38
Freightliner         7
Name: count, dtype: int64

38

Son 38 diferentes marcas o fabricantes del vehiculo

In [9]:
#conteo de datos categoricos:

conteoModel = dataTraining['Model'].value_counts()

display(conteoModel)
len(conteoModel)

Model
Silverado          18085
Grand              12344
Sierra              8409
Accord              7357
F-1504WD            6684
                   ...  
PathfinderSE          53
Galant4dr             53
SLK-ClassSLK350       52
Monte                 48
RX-84dr               48
Name: count, Length: 525, dtype: int64

525

Existen 525 diferentes tipos de modelos de vehiculos

Para tratar estos datos se realizara dummy y se eliminaran las variables categoricas originales, en total quedaran 617 variables binarias para dataTraining donde 1 quiere decir que si pertenece y cero que no. tambien, se tratara igual en conjunto de test.

In [10]:
#dummies
dataTraining = dataTraining.join(pd.get_dummies(dataTraining['Model'], prefix='M'))
dataTesting = dataTesting.join(pd.get_dummies(dataTesting['Model'], prefix='M'))
dataTraining = dataTraining.join(pd.get_dummies(dataTraining['Make'], prefix='K'))
dataTesting = dataTesting.join(pd.get_dummies(dataTesting['Make'], prefix='K'))
dataTraining = dataTraining.join(pd.get_dummies(dataTraining['State'], prefix='S'))
dataTesting = dataTesting.join(pd.get_dummies(dataTesting['State'], prefix='S'))
dataTraining = dataTraining.drop(['Model'], axis=1)
dataTesting = dataTesting.drop(['Model'], axis=1)
dataTraining = dataTraining.drop(['Make'], axis=1)
dataTesting = dataTesting.drop(['Make'], axis=1)
dataTraining = dataTraining.drop(['State'], axis=1)
dataTesting = dataTesting.drop(['State'], axis=1)

In [11]:
dataTraining.head()

Price  Year  Mileage    M_1  M_15002WD  M_15004WD  M_1500Laramie  \
0  34995  2017     9913  False      False      False          False   
1  37895  2015    20578  False      False      False          False   
2  18430  2012    83716  False      False      False          False   
3  24681  2014    28729  False      False      False          False   
4  26998  2013    64032  False      False      False          False   

   M_1500Tradesman  M_200LX  M_200Limited  ...  S_ SD  S_ TN  S_ TX  S_ UT  \
0            False    False         False  ...  False  False  False  False   
1            False    False         False  ...  False  False  False  False   
2            False    False         False  ...  False  False   True  False   
3            False    False         False  ...  False  False  False  False   
4            False    False         False  ...  False  False  False  False   

   S_ VA  S_ VT  S_ WA  S_ WI  S_ WV  S_ WY  
0  False  False  False  False  False  False  
1  False  False  False  False  False  False  
2  False  False  False  False  False  False  
3  False  False  False  False  False  False  
4  False  False  False  False  False  False  

[5 rows x 617 columns]

In [12]:
#Binaria
dataTraining = dataTraining.astype(int)
dataTesting= dataTesting.astype(int)
display(dataTraining.head())
display(dataTesting.head())

Price  Year  Mileage  M_1  M_15002WD  M_15004WD  M_1500Laramie  \
0  34995  2017     9913    0          0          0              0   
1  37895  2015    20578    0          0          0              0   
2  18430  2012    83716    0          0          0              0   
3  24681  2014    28729    0          0          0              0   
4  26998  2013    64032    0          0          0              0   

   M_1500Tradesman  M_200LX  M_200Limited  ...  S_ SD  S_ TN  S_ TX  S_ UT  \
0                0        0             0  ...      0      0      0      0   
1                0        0             0  ...      0      0      0      0   
2                0        0             0  ...      0      0      1      0   
3                0        0             0  ...      0      0      0      0   
4                0        0             0  ...      0      0      0      0   

   S_ VA  S_ VT  S_ WA  S_ WI  S_ WV  S_ WY  
0      0      0      0      0      0      0  
1      0      0      0      0      0      0  
2      0      0      0      0      0      0  
3      0      0      0      0      0      0  
4      0      0      0      0      0      0  

[5 rows x 617 columns]

Year  Mileage  M_1  M_15002WD  M_15004WD  M_1500Laramie  M_1500Tradesman  \
ID                                                                             
0   2014    31909    0          0          0              0                0   
1   2017     5362    0          0          0              0                0   
2   2014    50300    0          0          0              0                0   
3   2004   132160    0          0          0              0                0   
4   2015    25226    0          0          0              0                0   

    M_200LX  M_200Limited  M_200S  ...  S_ SD  S_ TN  S_ TX  S_ UT  S_ VA  \
ID                                 ...                                      
0         0             0       0  ...      0      0      0      0      0   
1         0             0       0  ...      0      0      0      0      0   
2         0             0       0  ...      0      0      0      0      0   
3         0             0       0  ...      0      0      0      0      0   
4         0             0       0  ...      0      0      0      0      0   

    S_ VT  S_ WA  S_ WI  S_ WV  S_ WY  
ID                                     
0       0      0      0      0      0  
1       0      0      0      0      0  
2       0      0      0      0      0  
3       0      1      0      0      0  
4       0      0      0      0      0  

[5 rows x 615 columns]

In [13]:
dataTraining.shape

(400000, 617)

Se nota que una diferencia de 2 entre entrentamiento y test cuando deberia ser solo una, el precio, se procede a verificar:

In [14]:
#comparativo

columnasTr=dataTraining.columns
columnasTs=dataTesting.columns

columnas_no_encontradas = []

for columna in columnasTr:
    if columna not in columnasTs:
        columnas_no_encontradas.append(columna)

print("Columnas en dataTraining que no están en dataTesting:")
print(columnas_no_encontradas)

Columnas en dataTraining que no están en dataTesting:
['Price', 'K_Freightliner']


A este punto en pro de mantener la consistencia entre dataTraining y dataTesting se eliminará la columna "K_Freightliner" del conjunto de entrenamiento g

In [15]:
dataTraining = dataTraining.drop(['K_Freightliner'], axis=1)
dataTraining

Price  Year  Mileage  M_1  M_15002WD  M_15004WD  M_1500Laramie  \
0       34995  2017     9913    0          0          0              0   
1       37895  2015    20578    0          0          0              0   
2       18430  2012    83716    0          0          0              0   
3       24681  2014    28729    0          0          0              0   
4       26998  2013    64032    0          0          0              0   
...       ...   ...      ...  ...        ...        ...            ...   
399995  29900  2015    25287    0          0          0              0   
399996  17688  2015    17677    0          0          0              0   
399997  24907  2014    66688    0          0          0              0   
399998  11498  2014    37872    0          0          0              0   
399999  16900  2014    78606    0          0          0              0   

        M_1500Tradesman  M_200LX  M_200Limited  ...  S_ SD  S_ TN  S_ TX  \
0                     0        0             0  ...      0      0      0   
1                     0        0             0  ...      0      0      0   
2                     0        0             0  ...      0      0      1   
3                     0        0             0  ...      0      0      0   
4                     0        0             0  ...      0      0      0   
...                 ...      ...           ...  ...    ...    ...    ...   
399995                0        0             0  ...      0      0      1   
399996                0        0             0  ...      0      0      0   
399997                0        0             0  ...      0      0      0   
399998                0        0             0  ...      0      0      0   
399999                0        0             0  ...      0      0      0   

        S_ UT  S_ VA  S_ VT  S_ WA  S_ WI  S_ WV  S_ WY  
0           0      0      0      0      0      0      0  
1           0      0      0      0      0      0      0  
2           0      0      0      0      0      0      0  
3           0      0      0      0      0      0      0  
4           0      0      0      0      0      0      0  
...       ...    ...    ...    ...    ...    ...    ...  
399995      0      0      0      0      0      0      0  
399996      0      0      0      0      0      0      0  
399997      0      0      0      0      0      0      0  
399998      0      0      0      0      0      0      0  
399999      0      0      0      0      0      0      0  

[400000 rows x 616 columns]

In [16]:
#eliminacion de caracteres problematicos en el nombre de las columans

import re

# Función para limpiar los nombres de las columnas
def limpia_columna_nom(column_name):
    # Reemplazar todos los caracteres no alfanuméricos con un guión bajo
    return re.sub(r'\W', '_', column_name)

# Aplicar la función a cada nombre de columna en dataTraining
dataTraining.columns = [limpia_columna_nom(col) for col in dataTraining.columns]
display(dataTraining.head())

dataTraining.head()


# Aplicar la función a cada nombre de columna en dataTraining
dataTesting.columns = [limpia_columna_nom(col) for col in dataTesting.columns]
display(dataTesting.head())

Price  Year  Mileage  M_1  M_15002WD  M_15004WD  M_1500Laramie  \
0  34995  2017     9913    0          0          0              0   
1  37895  2015    20578    0          0          0              0   
2  18430  2012    83716    0          0          0              0   
3  24681  2014    28729    0          0          0              0   
4  26998  2013    64032    0          0          0              0   

   M_1500Tradesman  M_200LX  M_200Limited  ...  S__SD  S__TN  S__TX  S__UT  \
0                0        0             0  ...      0      0      0      0   
1                0        0             0  ...      0      0      0      0   
2                0        0             0  ...      0      0      1      0   
3                0        0             0  ...      0      0      0      0   
4                0        0             0  ...      0      0      0      0   

   S__VA  S__VT  S__WA  S__WI  S__WV  S__WY  
0      0      0      0      0      0      0  
1      0      0      0      0      0      0  
2      0      0      0      0      0      0  
3      0      0      0      0      0      0  
4      0      0      0      0      0      0  

[5 rows x 616 columns]

Year  Mileage  M_1  M_15002WD  M_15004WD  M_1500Laramie  M_1500Tradesman  \
ID                                                                             
0   2014    31909    0          0          0              0                0   
1   2017     5362    0          0          0              0                0   
2   2014    50300    0          0          0              0                0   
3   2004   132160    0          0          0              0                0   
4   2015    25226    0          0          0              0                0   

    M_200LX  M_200Limited  M_200S  ...  S__SD  S__TN  S__TX  S__UT  S__VA  \
ID                                 ...                                      
0         0             0       0  ...      0      0      0      0      0   
1         0             0       0  ...      0      0      0      0      0   
2         0             0       0  ...      0      0      0      0      0   
3         0             0       0  ...      0      0      0      0      0   
4         0             0       0  ...      0      0      0      0      0   

    S__VT  S__WA  S__WI  S__WV  S__WY  
ID                                     
0       0      0      0      0      0  
1       0      0      0      0      0  
2       0      0      0      0      0  
3       0      1      0      0      0  
4       0      0      0      0      0  

[5 rows x 615 columns]

In [17]:
#divicion de de datos en conjunto de entrenamiento y prueba y predictores y variables objetivo

In [18]:
# Separación de variables predictoras (X) y variable de interés (y)
y = dataTraining['Price']
X = dataTraining.drop(['Price'], axis=1)

In [19]:
# descripcion del entrenamiento X
import pandas as pd
#formato de numero
pd.set_option('display.float_format', '{:.2f}'.format)

X.describe()

Year    Mileage       M_1  M_15002WD  M_15004WD  M_1500Laramie  \
count 400000.00  400000.00 400000.00  400000.00  400000.00      400000.00   
mean    2013.20   55072.96      0.00       0.00       0.00           0.00   
std        3.29   40881.02      0.02       0.03       0.05           0.03   
min     1997.00       5.00      0.00       0.00       0.00           0.00   
25%     2012.00   25841.00      0.00       0.00       0.00           0.00   
50%     2014.00   42955.00      0.00       0.00       0.00           0.00   
75%     2016.00   77433.00      0.00       0.00       0.00           0.00   
max     2018.00 2457832.00      1.00       1.00       1.00           1.00   

       M_1500Tradesman   M_200LX  M_200Limited    M_200S  ...     S__SD  \
count        400000.00 400000.00     400000.00 400000.00  ... 400000.00   
mean              0.00      0.00          0.01      0.00  ...      0.00   
std               0.02      0.02          0.07      0.04  ...      0.04   
min               0.00      0.00          0.00      0.00  ...      0.00   
25%               0.00      0.00          0.00      0.00  ...      0.00   
50%               0.00      0.00          0.00      0.00  ...      0.00   
75%               0.00      0.00          0.00      0.00  ...      0.00   
max               1.00      1.00          1.00      1.00  ...      1.00   

          S__TN     S__TX     S__UT     S__VA     S__VT     S__WA     S__WI  \
count 400000.00 400000.00 400000.00 400000.00 400000.00 400000.00 400000.00   
mean       0.02      0.11      0.01      0.04      0.00      0.03      0.01   
std        0.15      0.32      0.11      0.20      0.04      0.16      0.12   
min        0.00      0.00      0.00      0.00      0.00      0.00      0.00   
25%        0.00      0.00      0.00      0.00      0.00      0.00      0.00   
50%        0.00      0.00      0.00      0.00      0.00      0.00      0.00   
75%        0.00      0.00      0.00      0.00      0.00      0.00      0.00   
max        1.00      1.00      1.00      1.00      1.00      1.00      1.00   

          S__WV     S__WY  
count 400000.00 400000.00  
mean       0.00      0.00  
std        0.04      0.03  
min        0.00      0.00  
25%        0.00      0.00  
50%        0.00      0.00  
75%        0.00      0.00  
max        1.00      1.00  

[8 rows x 615 columns]

Existen 40 mil observaciones y 615 columnas

Year: los autos en el conjunto de datos varían desde 1997 hasta 2018, el año promedio de fabricación es aproximadamente 2013. La distribución parece bastante normal, ya que la media y la mediana (50%) son cercanas.

Mileage: la distancia promedio recorrida es de aproximadamente 55,000 millas, pero hay una gran variabilidad, como lo indica la desviación estándar de aproximadamente 40,881 millas, también hay un valor máximo muy alto de más de 2 millones de millas y podria catalogarse como dato atipico, se permitira que los modelos le den manejo a este tipo de datos.

In [20]:
#division
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen 

In [22]:
#opcion LightGBM con metodo de envoltura


import lightgbm as lgb
from sklearn.metrics import mean_squared_error
from math import sqrt
from sklearn.metrics import r2_score

# Inicializar el estimador con LightGBM
estima = lgb.LGBMRegressor()

# Preparar el conjunto de datos en el formato de LightGBM
train_data = lgb.Dataset(X_train, label=y_train)
valid_data = lgb.Dataset(X_test, label=y_test, reference=train_data)

# Configurar los parámetros del modelo
paramsLG = {
    'boosting_type': 'gbdt',
    'objective': 'regression',
    'metric': 'rmse',
    'max_depth': 3,
    'min_child_weight': 0.02,
    'num_leaves': 64,
    'n_estimators': 13000,
    'reg_alpha':1,

    'learning_rate': 0.2,
    'feature_fraction': 1,
    'bagging_fraction': 1,
    'bagging_freq': 8,
    'verbose': -1
}

# Entrenar el modelo
gbm = lgb.train(paramsLG,
                train_data,
                num_boost_round=10000,
                valid_sets=valid_data)

# Predecir
y_pred = gbm.predict(X_test, num_iteration=gbm.best_iteration)

# Evaluar con MSE y RMSE
mse_lgb_rfe = mean_squared_error(y_test, y_pred)
rmse_lgb_rfe = sqrt(mse_lgb_rfe)
r2 = r2_score(y_test, y_pred)
print(f'Mean Squared Error (MSE) con LightGBM: {mse_lgb_rfe}')
print(f'Root Mean Squared Error (RMSE) con LightGBM: {rmse_lgb_rfe}')
print(f'R" con LightGBM: {r2}')

<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
c:\Users\WD\.conda\envs\VSCEnv\lib\site-packages\lightgbm\engine.py:172: UserWarning: Found `n_estimators` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


Mean Squared Error (MSE) con LightGBM: 11681315.104957476
Root Mean Squared Error (RMSE) con LightGBM: 3417.793894452601
R" con LightGBM: 0.8984668581842542


In [27]:
import optuna

<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()


In [31]:
train_data = lgb.Dataset(X_train, label=y_train)
valid_data = lgb.Dataset(X_test, label=y_test, reference=train_data)
def objective(trial):
    # Preparar el conjunto de datos en el formato de LightGBM

    param = {
        'metric': 'rmse', 
        'random_state': 48,
        'n_estimators': 20000,
        'reg_alpha': trial.suggest_loguniform('reg_alpha', 1e-3, 10.0),
        'reg_lambda': trial.suggest_loguniform('reg_lambda', 1e-3, 10.0),
        'colsample_bytree': trial.suggest_categorical('colsample_bytree', [0.3,0.4,0.5,0.6,0.7,0.8,0.9, 1.0]),
        'subsample': trial.suggest_categorical('subsample', [0.4,0.5,0.6,0.7,0.8,1.0]),
        'learning_rate': trial.suggest_categorical('learning_rate', [0.006,0.008,0.01,0.014,0.017,0.02]),
        'max_depth': trial.suggest_categorical('max_depth', [10,20,100]),
        'num_leaves' : trial.suggest_int('num_leaves', 1, 1000),
        'min_child_samples': trial.suggest_int('min_child_samples', 1, 300),
        'cat_smooth' : trial.suggest_int('min_data_per_groups', 1, 100)
    }
    
    model = lgb.LGBMRegressor(**param)

    model.fit(X_train,y_train,eval_set=[X_test,y_test])

    # Predecir
    y_pred = model.predict(X_test, num_iteration=gbm.best_iteration)
    
    rmse = (mean_squared_error(y_test, y_pred))
    
    
    return rmse

In [32]:
study = optuna.create_study(direction='minimize')
study.optimize(objective, n_trials=50)

[I 2024-04-25 20:05:56,541] A new study created in memory with name: no-name-2f267a86-e9fe-44d5-ac74-4398f93f5d16
C:\Users\WD\AppData\Local\Temp\ipykernel_33012\856657795.py:10: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use suggest_float(..., log=True) instead.
  'reg_alpha': trial.suggest_loguniform('reg_alpha', 1e-3, 10.0),
C:\Users\WD\AppData\Local\Temp\ipykernel_33012\856657795.py:11: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use suggest_float(..., log=True) instead.
  'reg_lambda': trial.suggest_loguniform('reg_lambda', 1e-3, 10.0),
[W 2024-04-25 20:05:56,654] Trial 0 failed with parameters: {'reg_alpha': 0.9970125122383845, 'reg_lambda': 1.8397071079751386, 'colsample_bytree': 0.9, 'subsample': 0.8, 'learning_rate': 0.008, 'max_depth': 20, 'n

KeyError: 0

In [ ]:
# calculo para KAGGLE

# Cargar los conjuntos de entrenamiento y prueba
train_data = dataTraining
test_data = dataTesting

# Preparar las variables independientes y la variable objetivo
X_train = train_data.drop(['Price'], axis=1)  # Suponiendo que 'V1' es tu variable objetivo y 'Id' la columna de identificación
y_train = train_data['Price']

# Configurar los parámetros del modelo
paramsLGB = {
    'boosting_type': 'gbdt',
    'objective': 'regression',
    'metric': 'rmse',
    'max_depth': 3,
    'min_child_weight': 0.02,
    'num_leaves': 64,
    'n_estimators': 13000,
    'reg_alpha':1,

    'learning_rate': 0.2,
    'feature_fraction': 1,
    'bagging_fraction': 1,
    'bagging_freq': 8,
    'verbose': -1
}
# Crear el objeto Dataset de LightGBM
lgb_train = lgb.Dataset(X_train, y_train)

# Entrenar el modelo con todo el conjunto de entrenamiento
gbm = lgb.train(paramsLGB, lgb_train, num_boost_round=10000)

# Predecir las etiquetas del conjunto de prueba

X_testkaggle = test_data # Asegúrate de aplicar el mismo preprocesamiento que al conjunto de entrenamiento, si fuera necesario
y_pred_testkaggle = gbm.predict(X_testkaggle, num_iteration=gbm.best_iteration)


submission_df = pd.DataFrame(y_pred_testkaggle, index=dataTesting.index, columns=['Price'])

# Guardar predicciones en formato exigido en la competencia de kaggle
submission_df.to_csv('GIO_subXG.csv', index_label='ID')
submission_df.head()




C:\Users\Lenovo\AppData\Roaming\Python\Python311\site-packages\lightgbm\engine.py:172: UserWarning: Found `n_estimators` in params. Will use it instead of argument
  _log_warning(f"Found `{alias}` in params. Will use it instead of argument")


Price
ID         
0  21046.04
1  37112.89
2  23595.27
3   7877.29
4  30596.07